mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java

```diff
@@ -64,7 +66,19 @@ protected void onHandleIntent(Intent intent) {
                     notificationId = Integer.parseInt(intent.getExtras().getString(EXTRA_NOTIFICATION_ID));
                 }
                 if ("notification".equals(intent.getExtras().getString("type"))) {
-                    sendNotification(intent.getExtras().getString(EXTRA_MSG), notificationId);
+                    //we use local time as fallback
+                    long timestamp = System.currentTimeMillis();
+                    if(intent.getExtras().containsKey(EXTRA_TIMESTAMP)){
+                        //in case timestamp is passed by cloud service in later versions
+                        timestamp = intent.getExtras().getLong(EXTRA_TIMESTAMP);
+                    }else if(intent.getExtras().containsKey(EXTRA_GCM_TIMESTAMP)){
```

Comment:
If you don't want to simplify as per comment below, please keep spaces before/after curly braces and after keywords such as `if`.